File: daq/slc/src/dqm/src/TimedGraph1C.cc
#include "TimedGraph1C.h"

#include "CharArray.h"

using namespace Belle2;

TimedGraph1C::TimedGraph1C() throw() : TimedGraph1()
{
  _data = new CharArray(0);
}

TimedGraph1C::TimedGraph1C(const TimedGraph1C& h) throw() : TimedGraph1(h)
{
  _data = new CharArray(h._data->size());
}

TimedGraph1C::TimedGraph1C(const std::string& name, const std::string& title,
                           int nbinx, double xmin, double xmax)
throw() : TimedGraph1(name, title, nbinx, xmin, xmax)
{
  _data = new CharArray(nbinx);
  _time_v.resize(nbinx);
}

TimedGraph1C::~TimedGraph1C() throw() {}

std::string TimedGraph1C::getDataType() const throw()
{
  return std::string("TGC");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/TimedGraph1C.java
package b2daq.dqm.core;

public class TimedGraph1C extends TimedGraph1 {

	public TimedGraph1C() {
		super();
		data = new CharArray();
	}

	public TimedGraph1C(String name, String title, int nbins, double xmin, double xmax) {
		super(name, title, nbins, xmin, xmax);
		data = new CharArray(nbins);
	}

	public String getDataType() {
		return "TGC";
	}

}
